### PR TITLE
[CLEANUP] Remove Globals Resolver

### PR DESCRIPTION
**Pull request by btecu**

### CODE REVIEW

**Comment by mixonic** [comment]
@btecu there was a failure in the Node tests here, and this PR will require a rebase as well.

**Comment by btecu** [comment]
@mixonic I might need some help with the Node tests failure if you have some time.

**Comment by mixonic** [comment]
@btecu the `setup-app` boilerplate, and many of the tests in `tests/node/` are using deprecate API. Normally this is not permitted in the Ember test suite, and a test will fail if a deprecation is present without expectation. However these tests run in their own realm. Without that enforcement, it seems like they have fallen far off the best-practice path.

My commit adds a resolver to the app, thus avoiding the default resolver (which after your removal is just a hard failure).

**Comment by btecu** [comment]
@mixonic thanks for the explanation and your help!

File: packages/@ember/engine/index.js [hunk 1]
```diff
@@ -417,15 +416,6 @@ Engine.reopenClass({
...
   },
 
-  /**
-    Set this to provide an alternate class to `DefaultResolver`
-
-    @deprecated Use 'Resolver' instead
-    @property resolver
-    @public
-  */
-  resolver: null,
```
Comment (mixonic) [inline packages/@ember/engine/index.js:427]: This is weird. I see it was added in https://github.com/btecu/ember.js/commit/7b0a7ecb5b9e863bded0094d8b7e5c1861e1d52f#diff-a7583c2d819fdabaff78778f93e5112adc888bc3026338522862f08ddfadb4abR364 (or https://github.com/emberjs/ember.js/pull/12685/files#diff-a7583c2d819fdabaff78778f93e5112adc888bc3026338522862f08ddfadb4abR377 in the PR) by @dgeb but I don't see any usage of it. What am I missing? It was added with a `@deprecated` flag in jsdoc, but I don't see what is firing the deprecation.

**Review by mixonic — COMMENTED** [review]
Left an item I'd appreciate another set of eyes on. I think this is all good.

File: packages/@ember/engine/index.js (hunk 1 above)
Comment (dgeb) [reply packages/@ember/engine/index.js:427]: > but I don't see what is firing the deprecation.

I wonder if the code firing the deprecation was removed prior to 3.0 without removing the deprecated prop? Regardless, it doesn't seem that the `resolver` class prop is referenced, so it seems safe to remove this now.

**Review by dgeb — APPROVED** [review]
Looks good to me. Thanks @btecu and @mixonic for seeing this through.

**Pull request merged**